fileblock/src/simplefs.rs

```diff
@@ -0,0 +1,25 @@
+// slightly arbitrary size of our "disk": 4GIB, broken into 4KiB blocks
+// 4 * 2.pow(30) / 4096 == 2Kib required to store bitmap
+const BITMAP_LEN: usize = 1048576_usize;
+
+/// disk block free list 
+/// persistent tracking of what blocks have been allocated.
+/// option 1: Bitmap! naive style
+struct  DiskBlockFreelist {
+    // keep an array of bits indicating whether a file
+    bits: [usize; BITMAP_LEN]
+}
+
+/// option 2: B-Tree!!
```

Comment:
Storing a pointer for each block on disk is _way_ too expensive. I'm not sure what the math on "disk counting" vs "disk storage space" would be on this design, but I imagine not good.

Reply:
yeah i'm interested in figuring this out because i don't see a way to require _less_ storage than a compact bitmap, but possibly faster lookup?

Reply:
A bitmap would have constant time lookups since you know where any given block lies in the bitmap. A btree has O(logn) lookups for a block indexed by location. The b-tree is an option for the actual data storage layer (looking up blocks of data belonging to a given file).